src/app/base/components/node/NodeDevicesTable/NodeDevicesTable.test.tsx

```diff
@@ -0,0 +1,296 @@
+import { describe } from "vitest";
+
+import NodeDevicesTable from "./NodeDevicesTable";
+
+import { HardwareType } from "@/app/base/enum";
+import urls from "@/app/base/urls";
+import { MachineSidePanelViews } from "@/app/machines/constants";
+import type { ControllerDetails } from "@/app/store/controller/types";
+import type { MachineDetails } from "@/app/store/machine/types";
+import { NodeDeviceBus } from "@/app/store/nodedevice/types";
+import type { RootState } from "@/app/store/root/types";
+import { NodeActions, NodeStatusCode } from "@/app/store/types/node";
+import * as factory from "@/testing/factories";
+import {
+  renderWithProviders,
+  screen,
+  waitFor,
+  userEvent,
+} from "@/testing/utils";
+
+describe("NodeDevicesTable", () => {
+  let state: RootState;
+  let machine: MachineDetails;
+  let controller: ControllerDetails;
+
+  beforeEach(() => {
+    machine = factory.machineDetails({ system_id: "abc123" });
+    controller = factory.controllerDetails({ system_id: "abc123" });
+    const networkDevice = factory.nodeDevice({
+      bus: NodeDeviceBus.PCIE,
+      hardware_type: HardwareType.Network,
+      node_id: machine.id,
+    });
+    const storageDevice = factory.nodeDevice({
+      bus: NodeDeviceBus.PCIE,
+      hardware_type: HardwareType.Storage,
+      node_id: machine.id,
+    });
+    state = factory.rootState({
+      nodedevice: factory.nodeDeviceState({
+        items: [networkDevice, storageDevice],
+      }),
+    });
+  });
+
+  describe("display", () => {
+    it("displays a loading component if pools are loading", async () => {
+      state.nodedevice.loading = true;
+      renderWithProviders(
+        <NodeDevicesTable bus={NodeDeviceBus.PCIE} node={machine} />,
+        {
+          state,
+        }
+      );
+
+      await waitFor(() => {
+        expect(screen.getByText("Loading...")).toBeInTheDocument();
+      });
+    });
+
+    describe("displays a message when rendering an empty list", () => {
```

Comment:
```suggestion
    describe("empty state", () => {
```

I think this currently sounds too much like a test, but just my opinion